src/patients/notes/NewNoteModal.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react'
+import { Modal, Alert, RichText, Label } from '@hospitalrun/components'
+import { useTranslation } from 'react-i18next'
+import Note from '../../model/Note'
+
+interface Props {
+  show: boolean
+  toggle: () => void
+  onCloseButtonClick: () => void
+  onSave: (note: Note) => void
+}
+
+const NewNoteModal = (props: Props) => {
+  const { show, toggle, onCloseButtonClick, onSave } = props
+  const { t } = useTranslation()
+  const [errorMessage, setErrorMessage] = useState('')
+  const [note, setNote] = useState({
+    noteDate: new Date(),
+    text: '',
+  })
+
+  const onFieldChange = (key: string, value: string | any) => {
+    setNote({
+      ...note,
+      [key]: value,
+    })
+  }
+
+  const onRichTextElementChange = (
```

Comment:
I think that `onNoteText` may be a better function name since it better describes the field that it is attached to.